tests/integration/testutils/testserver.go

```diff
@@ -114,7 +115,11 @@ func StartAuthenticatorTestFramework(t *testing.T, stopCh <-chan struct{}, setup
 		t.Fatal(err)
 	}
 
-	return adminClient, clientWithExecAuthenticator
+	return adminClient, clientWithExecAuthenticator, func() {
```

Comment:
wrap the tear down method returned by `StartTestServer` with a couple of things we need to clean up related to things we started here in this method